File: src/lib/api/chains.ts
/**
 * api/chains.ts — 自定义动作链的后端调用（X1 B2）。
 * 对应 Rust：data_store/chains.rs + commands/chains.rs。
 */

import { invoke } from "@tauri-apps/api/core";
import type { ChainStep } from "@/lib/chains/types";

/** 一条用户自定义链（与后端 ChainDef 对齐，serde camelCase） */
export interface ChainDef {
  /** 空串 = 新建（后端生成 uuid） */
  id: string;
  name: string;
  description: string;
  /** 步骤：与前端 ChainStep 同构（transformId / risk / label） */
  steps: ChainStep[];
  sortOrder?: number;
  createdAt?: string;
  updatedAt?: string;
}

export const chainList = () => invoke<ChainDef[]>("chain_list");
export const chainSave = (chain: ChainDef) => invoke<string>("chain_save", { chain });
export const chainDelete = (id: string) => invoke<void>("chain_delete", { id });
export const chainReorder = (ids: string[]) => invoke<void>("chain_reorder", { ids });

File: src-tauri/src/commands/chains.rs
//! 动作链命令层（X1 B2）。
//!
//! 仿 commands/ai.rs 的自定义动作 CRUD：list / save / delete / reorder。
//! 步骤引用的变换是否存在由前端校验（变换注册表在前端），后端只管存取与结构约束。

use crate::data_store::{ChainDef, DataStore};
use tauri::State;

#[tauri::command]
pub fn chain_list(store: State<DataStore>) -> Result<Vec<ChainDef>, String> {
    store.chains()
}

#[tauri::command]
pub fn chain_save(store: State<DataStore>, chain: ChainDef) -> Result<String, String> {
    store.chain_save(&chain)
}

#[tauri::command]
pub fn chain_delete(store: State<DataStore>, id: String) -> Result<(), String> {
    store.chain_delete(&id)
}

#[tauri::command]
pub fn chain_reorder(store: State<DataStore>, ids: Vec<String>) -> Result<(), String> {
    store.chains_reorder(&ids)
}
